kivy/core/window/_window_sdl2.pyx

```diff
@@ -406,13 +407,19 @@ cdef class _WindowSDL2Storage:
         elif mode == 'colorkey':
             sdl_window_mode.mode = ShapeModeColorKey
 
-        sdl_shape = IMG_Load(<bytes>shape.encode('utf-8'))
-        if not sdl_shape:
+        try:
+            core_image = image.Image.load(shape, nocache=True)
+        except:
```

Comment:
Please don't. There are way better ways to handle errors and if you necessarily need `try-except`, then handle the error explicitly by using it in the `except` clause.

Reply:
It's a proof of concept, I didn't bother structuring it properly, yet. I've only tried to recreate the previous logic flow so far.

So yes, agree completely.